### PR TITLE
 Enable implicit integer coercion for interop libraries

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
Also:
* Improve code using interop
* Don't use fields for non-strict interop enum constants

### CODE REVIEW

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/StubGenerator.kt [hunk 1]
```diff
@@ -732,32 +732,31 @@ class StubGenerator(
...
-            val signedLiteral = integerLiteral(size, true, value)
-            val converter = when (size) {
-                1 -> "toUByte"
-                2 -> "toUShort"
-                4 -> "toUInt"
-                8 -> "toULong"
-                else -> return null
+        return if (isSigned) {
+            if (value == Long.MIN_VALUE) {
+                return "${value + 1} - 1" // Workaround for "The value is out of range" compile error.
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/StubGenerator.kt]: :)

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt [hunk 2]
```diff
@@ -674,6 +674,36 @@ internal class InteropLoweringPart1(val context: Context) : IrBuildingTransforme
         }
     }
 
+    override fun visitProperty(declaration: IrProperty): IrStatement {
+        val backingField = declaration.backingField
+        return if (declaration.isConst && backingField?.isStatic == true && context.config.isInteropStubs) {
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: Maybe comment why this is needed?

File: backend.native/tests/interop/basics/opengl_teapot.kt [hunk 3]
```diff
@@ -45,13 +45,13 @@ fun display() {
 
 fun initialize() {
     // select projection matrix
-    glMatrixMode(GL_PROJECTION.convert())
+    glMatrixMode(GL_PROJECTION)
```
Comment (olonho) [inline backend.native/tests/interop/basics/opengl_teapot.kt]: Cool!

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/StubGenerator.kt (hunk 1 above)
Comment (SvyatoslavScherbina) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/StubGenerator.kt]: And Kotlin still don't provide a way to define `Long.MIN_VALUE` with a single literal, so this workaround is still required!

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt (hunk 2 above)
Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: What do you mean by "this"? The entire transformation or the check above?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: The whole transformation.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: It is just an optimization and shouldn't be required by our memory model. Does it still need any comments?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: It better should be commented anyway, as this is pretty nontrivial transformation.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: Done?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt]: Yep

**Pull request merged**